Build-FFmpeg.ps1

```diff
@@ -0,0 +1,137 @@
+param(
+
+    [ValidateSet('x86', 'x64', 'ARM', 'ARM64')]
+    [string[]] $Platforms = ('x86', 'x64', 'ARM', 'ARM64'),
+
+    <#
+        Example values:
+        14.1
+        14.2
+        14.16
+        14.16.27023
+        14.23.27820
+    #>
+    [version] $VcVersion = '14.1',
+
+    <#
+        Example values:
+        8.1
+        10.0.15063.0
+        10.0.17763.0
+        10.0.18362.0
+    #>
+    [version] $WindowsTargetPlatformVersion = '10.0.15063.0',
+
+    [ValidateSet('Debug', 'Release')]
+    [string] $Configuration = 'Release',
+
+    [System.IO.DirectoryInfo] $VSInstallerFolder = "${env:ProgramFiles(x86)}\Microsoft Visual Studio\Installer",
+
+    [System.IO.FileInfo] $Msys2Bin = 'C:\msys64\usr\bin\bash.exe'
+)
+
+[System.IO.DirectoryInfo] $vsLatestPath = & "$VSInstallerFolder\vswhere.exe" -latest -property installationPath -products * -requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64

```

Comment:
We could add a "VisualStudioVersion" parameter to the PS script and then use e.g. `-version "[15.0,16.0]"` (that means >= 15.0 but < 16.0) for VS2017, `-version "[16.0,17.0]"` for VS2019, instead of using latest. Then we could select the complete build toolchain, not only SDK and PlatformToolkit.

Reply:
There is no 1:1 correspondence between VS versions, platform toolsets or Windows SDKs.

Especially with VS 2019, you could have very different combinations (see sample values in the parameters section).

While i'm using the most reasonable defaults I can think of, there is no reason why toolset `v141` can't be used with the latest WinSDK, or `v142` with the oldest one.

Reply:
What I mean is, currently you use vswhere -latest to get the VS folder. This means that it is not possible to build with VS2017 if you also have VS2019 installed. I would like to be able to select which VS version will be used to perform the build. So if I install VSNext, I can still build with the previous version.

We could also have a "VSVersionSelector" parameter, which is "-latest" by default, but can be set to e.g. "-version '[16.0,17.0]'" to force it to use VS2019.

Reply:
Resolved.